src/shogun/base/init.cpp

```diff
@@ -223,4 +232,37 @@ namespace shogun
 		}
 #endif
 	}
+
+	uint32_t generate_seed()
```

Comment:
we can actually replace this complicated function with a simple c++11 class `std::random_device`... basically:
```
uint32_t generate_seed()
{
    return std::random_device()();
}
```